File: backend/api/serializers/request.py
# serializers/request_serializers.py
from rest_framework import serializers
from ..models import Request

class RequestSerializer(serializers.ModelSerializer):
    class Meta:
        model = Request
        fields = ['id', 'user', 'volunteer', 'description', 'status', 'created_at', 'updated_at']
        read_only_fields = ['user', 'created_at', 'updated_at']

File: frontend/src/app/models/request.ts
import { User } from "./user"
import { Volunteer } from "./volunteer"

export interface Rekvest {
    "id": number,
    "user": number,
    "volunteer": number | null,
    "description": string,
    "status": string,
    "created_at": Date,
    "updated_at": Date
}

export interface RekvestExtended {
    "id": number,
    "user": User,
    "volunteer": Volunteer | null,
    "description": string,
    "status": string,
    "created_at": Date,
    "updated_at": Date
}